src/render/draw_background.js

```diff
@@ -27,9 +28,9 @@ function drawBackground(painter: Painter, sourceCache: SourceCache, layer: Backg
     const pass = (!image && color.a === 1 && opacity === 1) ? 'opaque' : 'translucent';
     if (painter.renderPass !== pass) return;
 
-    context.stencilTest.set(false);
-
-    painter.setDepthSublayer(0);
+    context.setStencilMode(StencilMode.disabled());
+    context.setDepthMode(painter.depthModeForSublayer(0, true));
```

Comment:
Oh — actually both are wrong. Since we draw opaque backgrounds in the opaque pass we should be writing to the depth mask only on that pass, as `drawFill` does. (I happened to try a test case with two opaque background layers, and they're rendered in the wrong order — only the bottom layer appears. Obviously this is an edge case that probably very few people would ever need to support, but 🤷‍♀️)